File: compiler/src/feedback/items/ConstantFeedback.swift
func registerConstantFeedback(into context: FeedbackContext) {
    context.register(
        id: "missing-constant-value",
        query: Queries.fact(MissingConstantValue.self),
        rank: { _, _ in .syntax },
        location: { node, _ in (primary: node, secondary: []) },
    ) { writer, node, _ in
        writer.write(node)
        writer.write(" is missing a value.")
        writer.writeBreak()
        writer.write("Try defining a value for this constant using ")
        writer.write(code: ":")
        writer.write(" on the following line.")
    }
}

File: compiler/feedback/src/items/constant_feedback.rs
use crate::{FeedbackCtx, FeedbackRank};
use wipple_queries::missing_constant_value;

pub fn register(ctx: &mut FeedbackCtx<'_>) {
    ctx.feedback("missing-constant-value")
        .query(missing_constant_value)
        .rank(|_| FeedbackRank::Syntax)
        .display(|_db, writer, node, _| {
            writer.node(node);
            writer.string(" is missing a value.");
            writer.line_break();
            writer.string("Try defining a value for this constant using ");
            writer.code(":");
            writer.string(" on the following line.");
        })
        .register();
}
